sitemap_utils.py

```
@@ -0,0 +1,367 @@
+        Args:
+            url: The URL to request
+            
+        Returns:
+            The response text or None if failed
+        """
+        for attempt in range(self.max_retries):
+            try:
+                self.throttler.throttle()
+                response = self.session.get(url, timeout=self.timeout)
+                response.raise_for_status()
+                return response.text
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)